### DISCUSSION TITLE
[ch 05. 트랜잭션과 잠금] 테이블 락을 명시적으로 사용하는 특수한 상황

### DISCUSSION THREAD

**Discussion by 5uhwann** [Learning-Is-Vital-In-Development/23-6-RealMySQL#21] [Q&A]
p.162 테이블 락을 명시적으로 획득해야하는 특별한 경우는 무엇이 있는가?

**Comment by dojinyou** [comment]
테이블 락의 경우 해당 테이블을 이용하는 모든 작업을 블락하여 가용성이 몹시 떨어지고 사용자에게 좋지 못한 경험을 줄 수 있습니다. 
따라서 정말 특별히 테이블 전체를 제어해야 하는 경우에는 replication을 통해 복제된 데이터베이스를 만들고 작업을 한 뒤에 작업 동안 변경된 작업을 로그로 가져와서 추가 반영해주는 것으로 알고 있습니다!
